### ISSUE TITLE
Where can I get training files？

### ISSUE THREAD

**Issue by G-rookie** [Yussef93/KnowWhatToLabel#2]
Thanks for your code, but this code does not provide the training file. Where can I get the training file?

**Comment by Yussef93** [comment]
Hi, Thanks for your interest, if you're referring to the source pre-training, we rely on the pre-trained weights so no need to train from scratch, however the source pre-training method "meta-learning" can be obtained from https://github.com/Yussef93/FewShotCellSegmentation

**Comment by EvgeniaChroni** [comment]
I tried to run the code but it has bugs ! I fixed it but the IOU that I get is 10 for TNBC which is much less to what you report. Could you please help us figure out what's going on? e.g. which are the parameter's values?

**Comment by Yussef93** [comment]
Hello,

Can you please share with me more details about the steps you did ?

**Comment by EvgeniaChroni** [comment]
I preprocesses and created the dataset by using your files. 

And the I run python Learning_main.py 
Below can be found the arguments of the Learning_main file 



`"""
Usage Instructions:

To meta train
ex:
python Learning_main.py --lr-method 'Meta_Learning' --architect 'FCRN' --target 'TNBC' --meta-train-losses 'BCE' --metalr 1.0 --lr 0.001 --pretrain-name 'test'

To train using supervised
ex:
python Learning_main.py --lr-method 'Supervised_Learning' --architect 'FCRN' --target 'TNBC' --lr 0.001 --pretrain-name 'test'
"""

import sys
import os
import argparse
sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
from Code import Meta_Learning,Supervised_Learning

def addLearningArgs():

    parser = argparse.ArgumentParser(description="Train with a Learning method")
    parser.add_argument('--lr_method', type=str, default='Meta_Learning', help="Enter Meta_Learning or Supervised_Learning")
    parser.add_argument("--bceloss", type=str, default="weightedbce",
                        help="standard BCE Loss function or weighted BCE Loss")
    parser.add_argument('--savedir', type=str, default='/Pre-trained/',
                        help="Enter Meta_Learning or Supervised_Learning")
    parser.add_argument('--datasets', type=str, nargs="*", default=['B5', 'B39', 'EM', 'ssTEM', 'TNBC'],
                        help="Combination of B5,B39,TNBC,ssTEM,EM")
    parser.add_argument('--architect', type=str, default='FCRN', help="Enter FCRN or UNet")
    parser.add_argument('--meta_train_losses', type=str, nargs="*", #['BCE', 'BCE_Entropy', 'BCE_Distillation', 'Combined']
                        default=['BCE'],
                        help="Combination of BCE,BCE_Entropy,BCE_Distillation,Combined")
    parser.add_argument('--meta_epochs', type=int, default=300)
    parser.add_argument('--meta_lr', type=float, default=1.0)
    parser.add_argument('--inner_epoch', type=int, default=20, help="# of Inner epochs in meta-training")
    parser.add_argument('--lr', type=float, default=0.001)

    parser.add_argument("--affine", type=int, default=0)
    parser.add_argument('--alpha',type=float,default=0.1)
    parser.add_argument('--beta',type=float,default=0.1)

    parser.add_argument('--epochs', type=int, default=100, help="# of Training epochs in supervised-learning")
    parser.add_argument('--target', type=str, nargs="*", default=['TNBC'],
                        help="Define Target dataset in leave-out-one-dataset cross validation")

    parser.add_argument('--name',type=str,default='test')

    return parser
def checkmetamethods(metatrainlosses_args,metatrainlosses):

    print(" metatrainlosses_args ", metatrainlosses_args)
    for metatrainloss in metatrainlosses_args:
        if metatrainloss not in metatrainlosses:
            raise ValueError(metatrainloss+ " is undefined")

def main():  

    parser = addLearningArgs()
    args = parser.parse_args()
    print(args)
    os.environ["CUDA_VISIBLE_DEVICES"] = "7"
    # checkmetamethods(args.meta_train_losses, metatrainlosses=['BCE', 'BCE_Entropy', 'BCE_Distillation', 'Combined'])

    if args.lr_method=='Meta_Learning':
        hyperparams = {'meta_lr': args.meta_lr,
                       'meta_epochs': args.meta_epochs,
                       'model_lr': args.lr,
                       'inner_epochs': args.inner_epoch,
                       'alpha': args.alpha,
                       'beta': args.beta,
                       'k-shot': 1,
                       'optimizer': {'weight_decay': 0.0005,
                                     'momentum': 0.9}}


        meta_learn = Meta_Learning.Meta_Learning(hyperparams=hyperparams, datasets=args.datasets,target_datasets=args.target,affine=bool(args.affine),
                                                 methods=args.meta_train_losses,architecture=args.architect,loss=args.bceloss,
                                                 experiment_name_postfix=args.name)

        meta_learn.meta_train()


    elif args.lrmethod=='Supervised_Learning':
        hyperparams = {'model_lr': args.lr,
                       'epochs': args.epochs,
                       'batchsize': 64,
                       'optimizer': {'weight_decay': 0.0005,
                                     'momentum': 0.9}
                       }


        datasets_path = './Datasets/FewShot/Source/'


        supervised_learn = Supervised_Learning.Supervised_Learning(hyperparams=hyperparams, datasets=args.datasets,
                                                                   save_dir=args.savedir,affine=bool(args.affine),
                                                                   datasets_path=datasets_path,loss=args.bceloss,
                                                                   targets=args.target, architecture=args.architect, experiment_name_postfix=args.name)

        supervised_learn.supervised_train()
    else:
        print("Learn Type is undefined")

if __name__ =='__main__':
    main()
`
